### PR TITLE
refactor(api): remove schema

### PR DESCRIPTION
**Pull request by ncclementi**
BREAKING CHANGE: Removed hierarchical usage of schema. 

Ibis uses the following naming conventions:
- schema: a mapping of column names to datatypes
- database: a collection of tables
- catalog: a collection of databases

### CODE REVIEW

**Comment by ncclementi** [comment]
I'm not sure I understand how the `test_signature.py` works, because I clearly broke all of them, but is failing due to an XPASS, any help on that front would be appreciated.

**Comment by cpcloud** [comment]
If something is xpassing, generally the approach is to delete the marker that's causing the problem.

**Comment by gforsyth** [comment]
the signature check is to enforce (eventually) that all the backends call all positional arguments the same thing, etc (that's why currently, almost all of them are xfailed)

When you removed the schema kwarg, it made some of the backend-specific table methods match the method definition from BaseBackend.  
also, keyword argument names also have to match, unless those arguments are keyword-only arguments

removing entries from those xfails is a good thing, though

File: ibis/backends/oracle/tests/test_client.py [hunk 1]
```diff
@@ -67,14 +67,11 @@ def stats_one_way_anova(x, y, value: str) -> int:
...
-def test_list_tables_schema_warning_refactor(con):
+def test_list_tables(con):
     assert con.list_tables()
 
     with pytest.raises(exc.IbisInputError):
-        con.list_tables(database="not none", schema="not none")
-
-    with pytest.warns(FutureWarning):
-        assert con.list_tables(schema="SYS", like="EXU8OPT") == ["EXU8OPT"]
+        con.list_tables(database="not none")
```
Comment (ncclementi) [inline ibis/backends/oracle/tests/test_client.py:74]: This seems to be broken, but I can't run the Oracle test suite locally, I'll need some help debugging this one.

File: ibis/backends/duckdb/tests/test_client.py [hunk 2]
```diff
@@ -78,16 +78,10 @@ def test_cross_db(tmpdir):
 
     con2.attach(path1, name="test1", read_only=True)
 
-    with pytest.warns(FutureWarning):
-        t1_from_con2 = con2.table("t1", schema="main", database="test1")
+    t1_from_con2 = con2.table("t1", database="test1")
```
Comment (gforsyth) [inline ibis/backends/duckdb/tests/test_client.py:81]:
```suggestion
    t1_from_con2 = con2.table("t1", database="test1.main")
```

File: ibis/backends/duckdb/tests/test_client.py [hunk 3]
```diff
@@ -78,16 +78,10 @@ def test_cross_db(tmpdir):
...
     assert t1_from_con2.schema() == t2.schema()
     assert t1_from_con2.execute().equals(t2.execute())
 
-    with pytest.warns(FutureWarning):
-        foo_t1_from_con2 = con2.table("t1", schema="foo", database="test1")
```
Comment (gforsyth) [inline ibis/backends/duckdb/tests/test_client.py:87]: This shouldn't be removed, but refactored to use either a dotted path or a tuple of path elements, like `"test1.foo"`

**Review by gforsyth — COMMENTED** [review]
Looks good overall,  i think we lost one bit of testing that we shouldn't have.

File: ibis/backends/trino/tests/test_client.py [hunk 4]
```diff
@@ -185,14 +185,6 @@ def test_list_tables_schema_warning_refactor(con):
...
     assert con.list_tables()
 
-    # Error if user mixes tuple inputs and string inputs for database and schema
-    with pytest.raises(FutureWarning):
-        with pytest.raises(exc.IbisInputError):
-            con.list_tables(database=("tuple", "ohstuff"), schema="str")
-
-    with pytest.warns(FutureWarning):
-        assert con.list_tables(database="tpch", schema="sf1") == tpch_tables
-
```
Comment (ncclementi) [inline ibis/backends/trino/tests/test_client.py:195]: @gforsyth Is this a similar case than the one pointed out above, about missing tests. Do we need to adapt this ones to pass `database="tpch.sf1"` I can't run the trino one locally to try them out

File: ibis/backends/duckdb/tests/test_client.py (hunk 3 above)
Comment (ncclementi) [reply ibis/backends/duckdb/tests/test_client.py:87]: fixed, I added it back using a dotted path.

File: ibis/backends/trino/tests/test_client.py (hunk 4 above)
Comment (gforsyth) [reply ibis/backends/trino/tests/test_client.py:195]: Yeah, no issues here because it's no longer possible to "overspecify" a path

**Review by gforsyth — APPROVED** [review]
Good riddance!

**Comment by gforsyth** [comment]
I've got to check to make sure the commit message is going to be formatted correctly, then we can merge

**Pull request merged**